runtime/Cpp/runtime/src/CommonToken.cpp

```diff
@@ -165,9 +164,15 @@ std::string CommonToken::toString(Recognizer *r) const {
   }
   std::string txt = getText();
   if (!txt.empty()) {
-    antlrcpp::replaceAll(txt, "\n", "\\n");
-    antlrcpp::replaceAll(txt, "\r", "\\r");
-    antlrcpp::replaceAll(txt, "\t", "\\t");
+    std::string escaped;
+    escaped.reserve(txt.size());  // Best case, nothing to escape.
+    for (const auto &c : txt) {
+      if (c == '\n' || c == '\r' || c == '\t') {
+        escaped.push_back('\\');
+      }
+      escaped.push_back(c);
+    }
+    txt = std::move(escaped);
```

Comment:
Is this really the better way? You now have duplicated the same code 3 times. Instead the existing code could be improved to do all replacements in one go. To be honest, I don't see an improvement in this patch.

Reply:
Switched back to keeping StringUtils.{h,cpp} and instead replaced `replaceAll` with `escapeWhitespace` which does the logic. I kept the same complexity by having two versions, one which returns a new std::string and another which accepts an std::string and appends.